### PR TITLE
[luci/lang] Introduce CircleVariable

### PR DESCRIPTION
**Pull request by seanshpark**
This will introduce CircleVariable virtual node to represent variable tensor.

ONE-DCO-1.0-Signed-off-by: SaeHie Park <saehie.park@gmail.com>

### CODE REVIEW

**Comment by seanshpark** [comment]
For #8203 , from draft #8279

**Review by mhs4670go — APPROVED** [review]
LGTM

**Review by BalyshevArtem — DISMISSED** [review]
LGTM

File: compiler/luci/lang/src/Nodes/CircleVariable.test.cpp [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  {
+  };
+
+  luci::CircleVariable var_node;
+
+  TestVisitor tv;
+  ASSERT_THROW(var_node.accept(&tv), std::exception);
+}
+
+TEST(CircleSVDFTest, visit_NEG)
```
Comment (seanshpark) [inline compiler/luci/lang/src/Nodes/CircleVariable.test.cpp:51]: This is copy-paste typo

**Review by mhs4670go — APPROVED** [review]

**Review by BalyshevArtem — APPROVED** [review]
LGTM

**Pull request merged**